File: bindings/lua/insight/signal/demod.lua
--- FM demodulation functions.
-- @module insight.signal.demod

local native = require("_insight")
local sig = native.signal

local M = {}

local _wrap = require("insight._wrap")

--- Demodulate an FM signal.
-- @tparam Array x Input FM signal array.
-- @int[opt=-1] axis Axis along which to demodulate. -1 is last axis.
-- @treturn Array Demodulated signal.
M.fm_demod = _wrap({ "x", "axis" }, function(x, axis)
  return sig.fm_demod(x, axis or -1)
end)

return M

File: bindings/python/insight/signal/demod.py
"""FM demodulation functions for signal processing.

Provides frequency demodulation of narrowband FM signals.
"""

from insight._insight import signal as _signal

__all__ = [
    "fm_demod",
]


def fm_demod(x, axis=-1):
    """Demodulate an FM signal.

    Estimates the instantaneous frequency of a narrowband FM signal by
    computing the derivative of the unwrapped phase.

    Args:
        x: Input complex-valued signal array.
        axis: Axis along which to demodulate.  Default is -1 (last axis).

    Returns:
        Real-valued demodulated signal (instantaneous frequency).
    """
    return _signal.fm_demod(x, axis=axis)
